### PR TITLE
add calico support for DC/OS

### PR DESCRIPTION
**Pull request by mainred**
## High-level description

As comments and conversation of the [earlier PR](https://github.com/dcos/dcos/pull/6244) is lengthly, I created this PR to give you a more simplified view to review.

Changes in this PR:
-  introduce Calico CNI plugin and calico-IPAM for Mesos UCR containers
-  add dcos-calico-node service per node as the control panel of Calico network
- refactor test_networking.py to included calico container network test
- add integration tests and e2e test for calico networking
- add vxlan support for calico in DC/OS

exposed configurations:

| Parameter            | Description                                                                                                                        |
|----------------------|------------------------------------------------------------------------------------------------------------------------------------|
| calico_enabled       | Indicates whether to enable Calico networking.[Default: 'true']                                                                                           |
| calico_network_cidr  | Subnet allocated for calico. The subnet specified by `calico_network_cidr` should not overlay with those for VXLAN backends or virtual networks defined for [DC/OS virtual networks](https://github.com/mesosphere/dcos-docs-site/blob/staging/pages/mesosphere/dcos/1.14/installing/production/advanced-configuration/configuration-reference/index.md#dcos_overlay_enable). Default: '192.168.0.0/16']                                                                          |
| calico_vxlan_enabled | Control, whether IP-in-IP or VXLAN mode is used for calico, by default IP-in-IP, is suggested to be used instead of VXLAN. `calico_vxlan_enabled` is supposed to set to 'true' for the environment that IP in IP is not supported, like Azure. [Default: 'true']    |
| calico_ipinip_mtu    | The MTU to set on the IPIP tunnel device. This configuration works when calico_vxlan_enabled is set to be false. Please refer to [this](https://docs.projectcalico.org/v3.8/networking/mtu) for a suitable MTU configuration. [Default: 1440]    |
| calico_vxlan_port    | The UDP port used for calico VXLAN. This configuration works when calico_vxlan_enabled is set to be true. [Default: 64000]           |
| calico_vxlan_vni     | The virtual network ID used for calico VXLAN. This configuration works when calico_vxlan_enabled is set to be true. [Default: 4096] |
| calico_vxlan_mtu     | The MTU to set on the VXLAN tunnel device. This configuration works when calico_vxlan_enabled is set to be true. Please refer to [this](https://docs.projectcalico.org/v3.8/networking/mtu) for a suitable MTU configuration [Default: 1410]    |
| calico_veth_mtu     | The MTU to set on the veth pairs created by calico CNI. Please refer to [this](https://docs.projectcalico.org/v3.8/networking/mtu) for a suitable MTU configuration [Default: 1410]    |

## Corresponding DC/OS tickets (required)

  - [DCOS-58582](https://jira.mesosphere.com/browse/DCOS-58582) calico plugin installation for Mesos UCR.
  - [DCOS-58583](https://jira.mesosphere.com/browse/DCOS-58583) Installation of Calico Control plane
  - [DCOS-59113](https://jira.mesosphere.com/browse/DCOS-59113) enable l4lb  of calico network container
  - [DCOS-58585](https://jira.mesosphere.com/browse/DCOS-58585) add e2e tests

## Related tickets (optional)

  - [DCOS-ID](https://jira.mesosphere.com/browse/DCOS-<number>) JIRA title / short description.

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain.
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]

### CODE REVIEW

File: gen/calc.py [hunk 1]
```diff
@@ -312,83 +313,122 @@ def validate_network_default_name(overlay_network_default_name, dcos_overlay_net
             overlay_network_default_name))
 
 
-def validate_dcos_ucr_default_bridge_subnet(dcos_ucr_default_bridge_subnet):
+class IPVersion(IntEnum):
+    """Defines the package source to install package from
```
Comment (ichernetsky) [inline gen/calc.py]: The docstring looks a bit off to me.

File: gen/calc.py [hunk 2]
```diff
@@ -312,83 +313,122 @@ def validate_network_default_name(overlay_network_default_name, dcos_overlay_net
...
+        if (version == IPVersion.IPv4 and not isinstance(network, ipaddress.IPv4Network)) or \
+                (version == IPVersion.IPv6 and not isinstance(network, ipaddress.IPv6Network)):
+            raise ValueError("IP version not match")
     except ValueError as ex:
-        raise AssertionError(
-            "Incorrect value for dcos_ucr_default_bridge_subnet: {}."
-            " Only IPv4 subnets are allowed".format(dcos_ucr_default_bridge_subnet)) from ex
+        err_msg = "Incorrect value for {}: {}. Only IPv{} subnets are allowed".format(
+            config_name, subnet, version)
+        raise AssertionError(err_msg) from ex
```
Comment (ichernetsky) [inline gen/calc.py]: Why `AssertionError` and not `ValueError` here? Is it an existing convention?

File: gen/calc.py [hunk 3]
```diff
@@ -312,83 +313,122 @@ def validate_network_default_name(overlay_network_default_name, dcos_overlay_net
...
+    validate_int_in_range(vtep_mtu, 552, None)
 
     for overlay in overlay_network['overlays']:
         assert 'name' in overlay, (
             'Missing "name" in overlay configuration: {}'.format(overlay))
 
-        assert (len(overlay['name']) <= 13), (
-            "Overlay name cannot exceed 13 characters:{}".format(overlay['name']))
+        assert (len(overlay['name']) <=
+                13), ("Overlay name cannot exceed 13 characters:{}".format(
```
Comment (ichernetsky) [inline gen/calc.py]: nit: a space after the colon.

File: gen/calc.py [hunk 4]
```diff
@@ -312,83 +313,122 @@ def validate_network_default_name(overlay_network_default_name, dcos_overlay_net
...
 
         assert ('subnet' in overlay or 'subnet6' in overlay), (
-            'Missing "subnet" or "subnet6" in overlay configuration:{}'.format(overlay))
-
-        assert 'vtep_mac_oui' in overlay_network.keys(), (
-            'Missing "vtep_mac_oui" in overlay configuration {}'.format(overlay_network))
-
-        vtep_mtu = overlay_network.get('vtep_mtu', 1500)
-        validate_int_in_range(vtep_mtu, 552, None)
+            'Missing "subnet" or "subnet6" in overlay configuration:{}'.format(
```
Comment (ichernetsky) [inline gen/calc.py]: ditto, in case you'll be updating this PR anyway. Otherwise, please ignore me.

File: gen/calc.py [hunk 5]
```diff
@@ -1067,6 +1107,20 @@ def node_check_names(config):
...
+    # dcos-net use fixed VXLAN UDP port 64000
+    # https://github.com/dcos/dcos-mesos-modules/blob/master/overlay/master.cpp#L1544
+    assert vxlan_port != 64000, 'VXLAN UDP port {} should not conflict with that of dcos-net'.format(vxlan_port)
+
+
+def validate_vxlan_vni(vxlan_vni):
+    validate_int_in_range(vxlan_vni, 4096, 16777215)
+    # dcos overlay use fixed VXLAN VNI 1024 assigned by the master module of dcos overlay network
+    # https://github.com/dcos/dcos-mesos-modules/blob/master/overlay/master.cpp#L1544
+    assert vxlan_vni != 1024, 'VXLAN VNI {} should not conflict with that of dcos overlay'
```
Comment (ichernetsky) [inline gen/calc.py]: This assertion is redundant, I believe

File: packages/dcos-integration-test/extra/test_networking.py [hunk 6]
```diff
@@ -340,11 +343,71 @@ def test_vip(dcos_api_session,
...
+    proxy container that will ping the origin container VIP and then assert
+    that the expected origin app UUID was returned
+    '''
+    if not lb_enabled():
+        pytest.skip('Load Balancer disabled')
+
+    errors = []
+    tests = setup_vip_workload_tests(dcos_api_session, container, vip_net,
+                                     proxy_net, ipv6, with_port_mapping_app, network_name="calico")
+    for vip, hosts, cmd, origin_app, proxy_app, pm_app in tests:
```
Comment (ichernetsky) [inline packages/dcos-integration-test/extra/test_networking.py]: `pm_app -> port_mapping_app` for better readability?

Comment (mainred) [reply packages/dcos-integration-test/extra/test_networking.py]: Hmm, I agree with you just from this test alone, and if you see other tests they use just `pm_app`.
Anyway, I found your trouble as a maintainer of this code, so let's improve the readability as your suggestion. Thanks.

File: gen/calc.py (hunk 5 above)
Comment (mainred) [reply gen/calc.py]: Oh, it's my mistake, you are correct.

File: CHANGES.md [hunk 7]
```diff
@@ -31,8 +31,13 @@ Please follow the [`CHANGES.md` modification guidelines](https://github.com/dcos
 * Port the Mesos open source metrics module to Windows. (DCOS-58008)
 
 * Add etcd into DC/OS. (DCOS-59004)
+
 * Update libpq to 9.6.15 (DCOS-59145)
 
+* Calico in DC/OS: introduced Calico network for mesos UCR. (DCOS-58582)
```
Comment (mainred) [inline CHANGES.md]: We can replace these multiple changes by just one item, `introducing calico in DC/OS`, added a TODO here: https://github.com/mainred/dcos/pull/2/files#r356544985

File: gen/calc.py [hunk 8]
```diff
@@ -1328,6 +1389,13 @@ def calculate_fault_domain_detect_contents(fault_domain_detect_filename):
         'enable_windows_agents': 'false',
         'windows_dcos_install_path': 'C:\\d2iq\\dcos',
         'windows_dcos_var_path': 'C:\\d2iq\\dcos\\var',
+        'calico_network_cidr': '192.168.0.0/16',
```
Comment (mainred) [inline gen/calc.py]: Unfortunately, we cannot calico configurations into only one blocker, because template analyzer does not support nested parameters.

Comment (mainred) [reply gen/calc.py]: `calico_network_cidr` should be a mandatory option, but as it takes too many changes including other reops like TeamCity configuration and terraform module update, so we tracked this task as a blocker here https://jira.mesosphere.com/browse/DCOS-60734

File: packages/conntrack-tools/build [hunk 9]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
```
Comment (mainred) [inline packages/conntrack-tools/build]: `conntrack-tools` contains `conntrack` command required by felix component to remove conntrack entries after workloads are deleted.

File: packages/dcos-integration-test/extra/test_networking.py [hunk 10]
```diff
@@ -83,11 +84,13 @@ def info(self, dcos_api_session):
...
 
     def hostport(self, dcos_api_session):
+        """ returns container ip and port for calico and dcos network otherwise the ones from the host
+        """
         info = self.info(dcos_api_session)
         task = info['app']['tasks'][0]
         if 'networks' in self.app and \
                 self.app['networks'][0]['mode'] == 'container' and \
-                self.app['networks'][0]['name'] == 'dcos':
+                self.app['networks'][0]['name'] in ["calico", "dcos"]:
```
Comment (mainred) [inline packages/dcos-integration-test/extra/test_networking.py]: This change is to make a minimal change to `test_networking.py` to influence other tests like `test_dcos_cni_l4lb`
https://github.com/dcos/dcos/pull/6582/files#diff-89643056102d7647724e7dc94be80843R793

File: packages/calico/extra/dcos-calico-felix.service [hunk 11]
```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=DC/OS Calico Node (felix): Calico Core component responsible for programming routes and security policy
+
+[Service]
+EnvironmentFile=/opt/mesosphere/environment
+EnvironmentFile=/opt/mesosphere/etc/calico/calico-node.env
+EnvironmentFile=/opt/mesosphere/etc/calico/calico-node-datastore.env
+ExecStartPre=/opt/mesosphere/bin/bootstrap dcos-calico-felix
+ExecStart=/opt/mesosphere/bin/start-calico-felix.sh
```
Comment (mainred) [inline packages/calico/extra/dcos-calico-felix.service]: By wrapping all the steps inside a script, we can also obtain the failed steps, and it helps me better maintain the code with comments in the script.

**Comment by mainred** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by mainred** [comment]
@mesosphere-mergebot request-review  @ichernetsky @vespian

File: CHANGES.md [hunk 12]
```diff
@@ -31,8 +31,13 @@ Please follow the [`CHANGES.md` modification guidelines](https://github.com/dcos
...
+
+* Calico in DC/OS: introduced calico control panel services deployed on all slaves and masters. (DCOS-58583)
```
Comment (vespian) [inline CHANGES.md]:
```suggestion
* Calico in DC/OS: introduced calico control panel services deployed on all agents and masters. (DCOS-58583)
```

File: gen/dcos-config.yaml [hunk 13]
```diff
@@ -3002,3 +3002,60 @@ package:
           Port 61092
           Chunk_Size 32
           Buffer_Size 64
+  - path: /etc/dcos/network/cni/calico.conf
```
Comment (vespian) [inline gen/dcos-config.yaml]: Collides with https://github.com/mesosphere/dcos-enterprise/pull/7006/files#diff-8acc8e77a15678f67a353d405fcf971cR1647

Comment (mainred) [reply gen/dcos-config.yaml]: We had an examples to define files in both gen/dcos-config.yaml and gen_extra/dcos-config.yaml
/etc_slave/telegraf/telegraf.d/agent.conf in [OSS](https://github.com/dcos/dcos/blob/master/gen/dcos-config.yaml#L2623) amd [EE](https://github.com/mesosphere/dcos-enterprise/blob/master/gen_extra/dcos-config.yaml#L1284)
and 
/etc/dcos-ui-update-service.env in [OSS](https://github.com/dcos/dcos/blob/master/gen/dcos-config.yaml#L2925) and [EE](https://github.com/mesosphere/dcos-enterprise/blob/master/gen_extra/dcos-config.yaml#L1638)

Let me double-check this.

File: packages/calico/extra/start-calico-felix.sh [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+set -xe
+
+export NODENAME=`/opt/mesosphere/bin/detect_ip`
+export IP=`/opt/mesosphere/bin/detect_ip`
+mkdir -p /etc/calico
+mkdir -p /var/lib/calico
+
+cp /opt/mesosphere/etc/calico/calicoctl.cfg /etc/calico/calicoctl.cfg
+cp /opt/mesosphere/active/calico/etc/profile.yaml /var/lib/calico/profile.yaml
+/opt/mesosphere/bin/calicoctl apply -f  /var/lib/calico/profile.yaml
```
Comment (vespian) [inline packages/calico/extra/start-calico-felix.sh]: Does it mean that if somebody removes our default policy and calico is restarted, the policy will be re-added?

File: packages/conntrack-tools/build [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+cp /usr/lib/x86_64-linux-gnu/libnetfilter_conntrack.so.3 $PKG_PATH/lib
+cp /usr/lib/x86_64-linux-gnu/libnfnetlink.so.0 $PKG_PATH/lib
+
+cd /pkg/src/conntrack-tools
+./configure --prefix="$PKG_PATH"
+
+make
+make install
+
+mv $PKG_PATH/sbin $PKG_PATH/bin
```
Comment (vespian) [inline packages/conntrack-tools/build]: Wouldn't it break things? Why do we need to do it?

File: packages/calico/extra/start-calico-felix.sh (hunk 14 above)
Comment (mainred) [reply packages/calico/extra/start-calico-felix.sh]: Yes. calicoctl handles input just like kubectl, and if you apply a definition, it will create/update this resource to now.

File: packages/conntrack-tools/build (hunk 15 above)
Comment (mainred) [reply packages/conntrack-tools/build]: To introduce `conntrack` binary.  I don't think this will break things. It is an augment change.

Comment (mainred) [reply packages/conntrack-tools/build]: Also as we introduce conntrack-tools only in the scope/env under DC/OS, so it will not be used by other software outside of DC/OS services.

File: gen/dcos-config.yaml (hunk 13 above)
Comment (mainred) [reply gen/dcos-config.yaml]: The behavior is confirmed. This is tricky.
In case we have the same file defined in both ` gen/dcos-config.yaml` and ` gen_extra/dcos-config.yaml`, these paths will be arranged in a list, and as the content of `gen_extra/dcos-config.yaml` is added later, so the file defined in `gen_extra/dcos-config.yaml` will show up later than the one defined in  `gen/dcos-config.yaml`, hence, when iterating the file path list , the later one will overwrite the first one.

Comment (mainred) [reply gen/dcos-config.yaml]: Here is an example:
gen/dcos-config.yaml
```
package:
  - path: /bin/detect_ip
    permissions: "0755"
```
gen_extra/dcos_config.yaml
```
package:
  - path: /bin/detect_ip
    permissions: "0777"
```

After rendering the templates, we'll have 
```
{'dcos-config.yaml': {'package': [{'path': '/bin/detect_ip', 'permissions': '0755'}, {'path': '/bin/detect_ip', 'permissions': '0777'}]}}
```

**Comment by mainred** [comment]
This PR relies on https://github.com/mesosphere/dcos-enterprise/pull/7095 because it helps fix EE bump PR CI.

**Comment by mainred** [comment]
@ichernetsky @vespian  could you please take another look?

File: gen/dcos-config.yaml (hunk 13 above)
Comment (ichernetsky) [reply gen/dcos-config.yaml]: 😲 Quite surprising.

File: gen/calc.py [hunk 16]
```diff
@@ -322,83 +323,119 @@ def validate_network_default_name(overlay_network_default_name, dcos_overlay_net
...
-                    "Incorrect value for overlay subnet6 {}."
-                    " Only IPv6 values are allowed".format(overlay_network['subnet6'])) from ex
+            validate_config_subnet("subnet6", overlay['subnet6'],
+                                   IPVersion.IPv6)
 
         if 'enabled' in overlay:
             gen.internals.validate_one_of(overlay['enabled'], [True, False])
 
 
+def validate_net_overlap(dcos_overlay_network, dcos_overlay_enable,
```
Comment (vespian) [inline gen/calc.py]:
```suggestion
def validate_overlay_networks_dont_overlap(dcos_overlay_network, dcos_overlay_enable,
```

File: gen/dcos-config.yaml (hunk 13 above)
Comment (vespian) [reply gen/dcos-config.yaml]: Thanks for double-checking it @mainred !

File: packages/calico/extra/dcos-calico-bird.service [hunk 17]
```diff
@@ -0,0 +1,16 @@
+[Unit]
```
Comment (vespian) [inline packages/calico/extra/dcos-calico-bird.service]: We will be always running an overlay network (vxlan/ipinip), right? Do we then still need to run bird on DC/OS?

File: packages/calico/extra/start-calico-felix.sh (hunk 14 above)
Comment (vespian) [reply packages/calico/extra/start-calico-felix.sh]: This would mean that it is not possible to remove our default policy and that our customers may get confused at times. Imagine a situation when somebody removes the default policy and then after 2-3 months the server gets restarted - the policy appears again. 

Is product ok with that?

**Comment by vespian** [comment]
@mainred What is the plan regarding the CI failures for EE:
```
[2019-12-12 13:23:18,987|dcos_e2e._subprocess_tools|DEBUG]: E         '172.17.0.4': {'dcos-adminrouter-agent.service',
[2019-12-12 13:23:18,987|dcos_e2e._subprocess_tools|DEBUG]: E         -                 'dcos-calico-bird.service',
[2019-12-12 13:23:18,987|dcos_e2e._subprocess_tools|DEBUG]: E         -                 'dcos-calico-confd.service',
[2019-12-12 13:23:18,987|dcos_e2e._subprocess_tools|DEBUG]: E         -                 'dcos-calico-felix.service',
```
Open DC/OS tests are run also in EE, and with no calico in EE yet, they will inevitably fail.

File: packages/calico/extra/start-calico-felix.sh (hunk 14 above)
Comment (mainred) [reply packages/calico/extra/start-calico-felix.sh]: Correct.
But we do not expose network policy functionality to our customers during this release, and  managing profile may be dangerous, this may be mentioned in calico doc, and calicoctl doc you are working on.

**Comment by mainred** [comment]
> @mainred What is the plan regarding the CI failures for EE:

@vespian  could you please take a look at This PR https://github.com/mesosphere/dcos-enterprise/pull/7095? This explains why. Thanks a lot.

**Comment by mainred** [comment]
@orlandohohmeier , please switch the base branch to the intermediate branch from master, thanks.

File: packages/dcos-integration-test/extra/test_networking.py [hunk 18]
```diff
@@ -340,11 +343,71 @@ def test_vip(dcos_api_session,
...
     assert not errors
 
+    containers = list(marathon.Container) + [Container.POD]
+    return [(container, vip_net, proxy_net)
+            for container in containers
+            for vip_net in list(marathon.Network)
+            for proxy_net in list(marathon.Network)]
+
+
+@pytest.mark.parametrize('container', [marathon.Container.MESOS, Container.POD])
```
Comment (vespian) [inline packages/dcos-integration-test/extra/test_networking.py]: This will be extended with DOCKER container, once the libnetwork support lands. Right?

File: packages/dcos-integration-test/extra/test_networking.py [hunk 19]
```diff
@@ -340,11 +343,71 @@ def test_vip(dcos_api_session,
...
+@pytest.mark.parametrize('vip_net', [marathon.Network.USER])
+@pytest.mark.parametrize('proxy_net', [marathon.Network.USER])
+def test_calico_vip(dcos_api_session,
+                    container: marathon.Container,
+                    vip_net: marathon.Network,
+                    proxy_net: marathon.Network,
+                    ipv6: bool=False,
+                    with_port_mapping_app=True):
+    '''Test VIPs between the following source and destination configurations:
+        * containers: DOCKER, UCR and NONE
```
Comment (vespian) [inline packages/dcos-integration-test/extra/test_networking.py]: Is this comment relevant to the current configuration of this test?

File: test-e2e/test_calico_networking.py [hunk 20]
```diff
@@ -0,0 +1,111 @@
...
+    unit_properties = result.stdout.strip().decode()
+    if active:
+        assert "ActiveState=active" in unit_properties
+    else:
+        assert "ActiveState=inactive" in unit_properties
+        assert "ConditionResult=no" in unit_properties
+
+
+@pytest.fixture(scope="module")
+def calico_ipip_cluster(docker_backend: Docker, artifact_path: Path,
```
Comment (vespian) [inline test-e2e/test_calico_networking.py]:
```suggestion
def calico_ipip_cluster(e2e_backend: Docker, artifact_path: Path,
```

File: gen/dcos-config.yaml [hunk 21]
```diff
@@ -2999,3 +2999,60 @@ package:
...
+    content: |
+      {
+        "name": "calico",
+        "type": "mesos-cni-port-mapper",
+        "excludeDevices": [
+          "cali+"
+        ],
+        "chain": "M-CALICO",
+        "delegate": {
+          "cniVersion": "0.1.0",
```
[the remark this answers is not in the document]
Comment (mainred) [reply gen/dcos-config.yaml]: I wanted to explain this.
Mesos do not support 0.3.1 totally, and 0.1.0 is enough to work.

File: gen/dcos-config.yaml [hunk 22]
```diff
@@ -2999,3 +2999,60 @@ package:
...
+          }
+        }
+      }
+  - path: /etc/calico/calico-node.env
+    content: |
+      NO_DEFAULT_POOLS=false
+      CALICO_IPV4POOL_CIDR={{ calico_network_cidr }}
+      CALICO_IPV4POOL_NAT_OUTGOING=true
+      FELIX_IGNORELOOSERPF=true
+      FELIX_LOGFILEPATH=none
```
[the remark this answers is not in the document]
Comment (mainred) [reply gen/dcos-config.yaml]: systemd journal

File: gen/dcos-config.yaml [hunk 23]
```diff
@@ -2999,3 +2999,60 @@ package:
...
+            "type": "calico-ipam"
+          }
+        }
+      }
+  - path: /etc/calico/calico-node.env
+    content: |
+      NO_DEFAULT_POOLS=false
+      CALICO_IPV4POOL_CIDR={{ calico_network_cidr }}
+      CALICO_IPV4POOL_NAT_OUTGOING=true
+      FELIX_IGNORELOOSERPF=true
```
[the remark this answers is not in the document]
Comment (mainred) [reply gen/dcos-config.yaml]: Pleaese read this https://dcosjira.atlassian.net/browse/DCOS-230

File: test-e2e/test_calico_networking.py (hunk 20 above)
Comment (mainred) [reply test-e2e/test_calico_networking.py]: Other backends are not recommended as the input. So let's focus on docker_backend

File: packages/dcos-integration-test/extra/test_networking.py (hunk 19 above)
Comment (mainred) [reply packages/dcos-integration-test/extra/test_networking.py]: copied from `test_vip`

Comment (mainred) [reply packages/dcos-integration-test/extra/test_networking.py]: To achieve minimal change.
`test_vip` is not allowed to extend calico tests, so I just copied `test_vip`.
Actually I tried to refactor the `test_networking.py` a bit to be more extensible, but we have too many bug fixes in this file.

File: test-e2e/test_calico_networking.py (hunk 20 above)
Comment (mainred) [reply test-e2e/test_calico_networking.py]: Sorry, I misunderstood here.
But we don't have an `e2e_backend` fixture.

File: packages/dcos-integration-test/extra/test_networking.py (hunk 18 above)
Comment (mainred) [reply packages/dcos-integration-test/extra/test_networking.py]: You got it. 
Adding libnetwork tests can be very handy, https://github.com/mainred/dcos/pull/1/files#diff-89643056102d7647724e7dc94be80843

File: gen/dcos-config.yaml (hunk 23 above)
[the remark this answers is not in the document]
Comment (vespian) [reply gen/dcos-config.yaml]: OK, so I understand that we require this for mesos-overlay to work, right? If so, once we deprecate mesos-overlay, will we be able to switch this to `false` and use `rp_filter=1` setting?

File: gen/dcos-config.yaml (hunk 21 above)
[the remark this answers is not in the document]
Comment (vespian) [reply gen/dcos-config.yaml]: Thank you for the detailed answer!  Do you think you could put it somewhere in the design doc so that it does not get lost?

File: packages/dcos-integration-test/extra/test_networking.py (hunk 19 above)
Comment (vespian) [reply packages/dcos-integration-test/extra/test_networking.py]: > copied from test_vip

What I meant is that the description reflects the parameters of the `test_vip` but the parameters of `test_calico_vip` are different.

File: test-e2e/test_calico_networking.py (hunk 20 above)
Comment (vespian) [reply test-e2e/test_calico_networking.py]: Ah, did not notice that this is a fixture, thought that this is just a param. Sorry for the confusion!

File: gen/dcos-config.yaml (hunk 22 above)
[the remark this answers is not in the document]
Comment (vespian) [reply gen/dcos-config.yaml]: So if the log is disabled, then calico is logging to stdout, right? Having a comment here would be a great help for folks like me trying to understand how it works :)

File: gen/dcos-config.yaml (hunk 23 above)
[the remark this answers is not in the document]
Comment (mainred) [reply gen/dcos-config.yaml]: Yes, exactly.

[the remark this answers is not in the document]
Comment (vespian) [reply gen/dcos-config.yaml]: Cool! Can we please make a comment here plus e.g. create Jira or a section in the design doc that describes all the steps required to deprecate mesos-overlay?

[the remark this answers is not in the document]
Comment (mainred) [reply gen/dcos-config.yaml]: Sure, thanks, a Jira is required. But I am afraid deprecating dcos overlay will be an epic, and I need also confirm if we already have this task in our roadmap.

File: gen/dcos-config.yaml (hunk 21 above)
[the remark this answers is not in the document]
Comment (mainred) [reply gen/dcos-config.yaml]: Done, thanks.

File: packages/dcos-integration-test/extra/test_networking.py (hunk 19 above)
Comment (mainred) [reply packages/dcos-integration-test/extra/test_networking.py]: I'd say the configurations are used to explain the combinations of scenarios we are going to test, not to explain the function parameter.
I will remove these not covered by this test.

File: packages/calico/extra/start-calico-felix.sh (hunk 14 above)
Comment (mainred) [reply packages/calico/extra/start-calico-felix.sh]: I corrected this by applying the default profile when a node is initialized and Calico profile does not exist.

Comment (mainred) [reply packages/calico/extra/start-calico-felix.sh]: Kubernetes achieves the default namespace-based network policy when a namespace is created, but we do not hold a concept like namespace.

File: gen/dcos-config.yaml (hunk 23 above)
[the remark this answers is not in the document]
Comment (mainred) [reply gen/dcos-config.yaml]: https://jira.mesosphere.com/browse/DCOS-62521

**Comment by mainred** [comment]
@vespian @ichernetsky could you please give another look?

**Comment by mainred** [comment]
@mesosphere-mergebot request-review  @wavesoft

**Comment by vespian** [comment]
@mainred I just noticed that this went into the `master` - I though that Calico changes were meant for the intermediate branch as we are targeting 2.2  for the release?

**Comment by mainred** [comment]
@vespian , yes, we were supposed to merge this PR into the intermediate branch, but we find it's pretty hard to have PRs on an intermediate branch to inherit the CIs we have for the master branch.
And as we are not going to support Calico in windows, and calico components are excluded in windows packages, so we decided to switch to the master branch.
CC @orlandohohmeier
